env.sh

```diff
@@ -7,11 +7,11 @@
 conda install -c conda-forge ffmpeg
 
 # Pip packages
-pip install setuptools ruamel.yaml tqdm colorama easydict tabulate loguru json5 Cython unidecode inflect argparse g2p_en tgt librosa matplotlib typeguard einops omegaconf hydra-core humanfriendly pandas
+pip install setuptools ruamel.yaml tqdm colorama easydict tabulate loguru json5 Cython unidecode inflect argparse g2p_en tgt librosa==0.9.1 matplotlib typeguard einops omegaconf hydra-core humanfriendly pandas
 
 pip install tensorboard tensorboardX torch==2.0.1 torchaudio==2.0.2 torchvision==0.15.2 accelerate==0.24.1 transformers diffusers praat-parselmouth audiomentations pedalboard ffmpeg-python==0.2.0 pyworld diffsptk nnAudio unidecode inflect ptwt
 
-pip install torchmetrics pymcd openai-whisper frechet_audio_distance asteroid resemblyzer
+pip install torchmetrics pymcd openai-whisper frechet_audio_distance asteroid resemblyzer vector-quantize-pytorch==1.12.5
```

Comment:
Is `vector-quantize-pytorch` a new package? Which part is dependent on it? Do we need it before?

Reply:
As mentioned above, `vector-quantize-pytorch` is a package installed when installing `diffsptk` at line 12. 

When running preprocessing stage for fs2 with a fresh amphion environment prepared with `env.sh`, the program gave the following error: `AttributeError: module 'torch' has no attribute 'compiler'`. However, when running with an  amphion environment installed before, no such error occurred.

After listing all package versions and comparing the differences, `vector-quantize-pytorch` stood out for its relevance to the error message and its change in name. Currently, the package manager fetches `vector_quantize_pytorch` 1.12.6 by default (notice the underscores instead of hyphens in the package name) instead of `vector-quantize-pytorch`, and causing errors in preprocessing of fs2. 

By reverting the version to `vector-quantize-pytorch` 1.12.5, the preprocessing ran without errors occurring.

Therefore, this addition of `vector-quantize-pytorch==1.12.5` is for reverting to an earlier version of the package after `diffsptk` installs the erroneous 1.12.6. 

This fix by specifying the version for this package to be no more than 1.12.5 until a change in package manager behavior is temporary in nature. At its core, specifying the version for `diffsptk` package may be a better solution for stableness in the long run.

FYI, the `diffsptk` package version for the older, error-free environment is 1.0.1. For the newer, error-occurred environment is 1.1.0.

Reply:
So do we need to specify the diffsptk's version? If this will be more stable, you can modify it.

Reply:
After testing, it turns out that simply specifying `diffsptk` version does not resolve the issue. The package manager still installs the latest version for `vector-quantize-python`, causing the error.

For now, the version for `vector-quantize-pytorch` needs to be specified.